ide/coqide/dune

```diff
@@ -26,11 +26,22 @@
 (library
  (name coqide_gui)
  (wrapped false)
- (modules (:standard \ document idetop coqide_main default_bindings_src gen_gtk_platform))
+ (modules (:standard \ document idetop coqide_main default_bindings_src gen_gtk_platform
+    shared shared_os_specific))
  (foreign_stubs
   (language c)
   (names coqide_os_stubs))
- (libraries coqide-server.protocol coqide-server.core lablgtk3-sourceview3))
+ (optional)
```

Comment:
I think this `optional` wasn't meant to be added back

Reply:
See #16672.

I am careful, but also fallible.

We could avoid some rebasing and resulting merge errors by reducing the long wait between writing the code and its merge (about 4 months for this fairly small PR).  That's long enough to forget many details.  It took me a couple hours over two days to remember how to build and retest the code on Windows.  (The weirdest detail being that I can't push from the cygwin/Windows command line because I couldn't figure out how to set up the security token in Windows--as it was last time.  I can only push from the WSL command line.)